components/gcp/bigquery/query/README.md

```diff
@@ -1,62 +1,104 @@
+dataset_location | The location to create the dataset. Defaults to `US`. | String | Yes | `US`
+job_config | The full config spec for the query job. See [QueryJobConfig](https://googleapis.github.io/google-cloud-python/latest/bigquery/generated/google.cloud.bigquery.job.QueryJobConfig.html#google.cloud.bigquery.job.QueryJobConfig) for details. | Dict | Yes | ` `
+
+
+## Outputs
+Name | Description | Type
+:--- | :---------- | :---
+output_gcs_path | The path to the Cloud Storage bucket containing the query output in CSV format. | GCSPath
+
+## Cautions and requirements
+To use the component, the following requirements must be met:
+* BigQuery API is enabled
+* The component is running under a secret of [Kubeflow user service account](https://www.kubeflow.org/docs/started/getting-started-gke/#gcp-service-accounts) in a Kubeflow cluster. For example:
 
-## Run-Time Parameters:
-Name | Description
-:--- | :----------
-query | The query used by Bigquery service to fetch the results.
-project_id | The project to execute the query job.
-dataset_id | The ID of the persistent dataset to keep the results of the query. If the dataset does not exist, the operation will create a new one.
-table_id | The ID of the table to keep the results of the query. If absent, the operation will generate a random id for the table.
-output_gcs_path | The GCS blob path to dump the query results to.
-dataset_location | The location to create the dataset. Defaults to `US`.
-job_config | The full config spec for the query job. See [QueryJobConfig](https://googleapis.github.io/google-cloud-python/latest/bigquery/generated/google.cloud.bigquery.job.QueryJobConfig.html#google.cloud.bigquery.job.QueryJobConfig) for details.
+```python
+bigquery_query_op(...).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
-## Output:
-Name | Description
-:--- | :----------
-output_gcs_path | The GCS blob path to dump the query results to.
+```
 
-## Sample
+* The Kubeflow user service account is a member of `roles/bigquery.admin` role of the project.
+* The Kubeflow user service account is also a member of `roles/storage.objectCreator` role of the Cloud Storage output bucket.
 
-Note: the sample code below works in both IPython notebook or python code directly.
+## Detailed Description
+The component does several things:
+1. Creates persistent dataset and table if they do not exist.
+1. Submits a query to BigQuery service and persists the result to the table.
+1. Creates an extraction job to output the table data to a Cloud Storage bucket in CSV format.
 
-### Set sample parameters
+Here are the steps to use the component in a pipeline:
+1. Install KFP SDK
+Install the SDK (Uncomment the code if the SDK is not installed before)
 
 
 ```python
-# Required Parameters
-PROJECT_ID = '<Please put your project ID here>'
-GCS_WORKING_DIR = 'gs://<Please put your GCS path here>' # No ending slash
+%%capture
```

Comment:
What is %%capture command?

Reply:
This is used to hide the outputs from the cell. It's usually not quite useful to show pip install logs here.

Reply:
Maybe we still need errors to be output.
https://ipython.readthedocs.io/en/stable/interactive/magics.html#cellmagic-capture shows a way to not capture stderr. 
however, when I tried it in my notebook with !pip install, it does not work.

Reply:
BTW, `pip` has the `–quiet` option.